api/client.go

```diff
@@ -292,9 +292,12 @@ func (c *Client) doRequest(req *http.Request, v any) (*Response, error) {
 			io.Copy(w, resp.Body)
 		} else {
 			if strings.Contains(req.Header.Get("Content-Type"), "application/msgpack") {
-				err = errors.New("Msgpack not supported")
+				return response, errors.New("Msgpack not supported")
 			} else {
 				err = json.NewDecoder(resp.Body).Decode(v)
+				if err != nil {
+					return response, fmt.Errorf("failed to decode JSON response: %v", err)
+				}
 			}
```

Comment:
```suggestion
		err = json.NewDecoder(resp.Body).Decode(v)
		if err != nil {
			return response, fmt.Errorf("failed to decode JSON response: %v", err)
		}
```
If you're returning, you may as well drop the else clause.